src/vm/debugdebugger.cpp

```diff
@@ -1367,59 +1367,70 @@ FCIMPL4(INT32, DebuggerAssert::ShowDefaultAssertDialog,
     
     int         result          = IDRETRY;
 
-    struct _gc {
-        STRINGREF strCondition;
-        STRINGREF strMessage;
-        STRINGREF strStackTrace;
-        STRINGREF strWindowTitle;
-    } gc;
-
-    gc.strCondition    = (STRINGREF) ObjectToOBJECTREF(strConditionUNSAFE);
-    gc.strMessage      = (STRINGREF) ObjectToOBJECTREF(strMessageUNSAFE);
-    gc.strStackTrace   = (STRINGREF) ObjectToOBJECTREF(strStackTraceUNSAFE);
-    gc.strWindowTitle  = (STRINGREF) ObjectToOBJECTREF(strWindowTitleUNSAFE);
-
-    HELPER_METHOD_FRAME_BEGIN_RET_PROTECT(gc);
-
-    StackSString condition;
-    StackSString message;
-    StackSString stackTrace;
-    StackSString windowTitle;
-
-    if (gc.strCondition != NULL)
-        gc.strCondition->GetSString(condition);
-    if (gc.strMessage != NULL)
-        gc.strMessage->GetSString(message);
-    if (gc.strStackTrace != NULL)
-        gc.strStackTrace->GetSString(stackTrace);
-    if (gc.strWindowTitle != NULL)
-        gc.strWindowTitle->GetSString(windowTitle);
-
-    StackSString msgText;
-    if (gc.strCondition != NULL) {
-        msgText.Append(W("Expression: "));
-        msgText.Append(condition);
-        msgText.Append(W("\n"));
-    }
-    msgText.Append(W("Description: "));
-    msgText.Append(message);
-    
-    StackSString stackTraceText;
-    if (gc.strStackTrace != NULL) {
-        stackTraceText.Append(W("Stack Trace:\n"));
-        stackTraceText.Append(stackTrace);
+#ifdef _DEBUG
```

Comment:
I think we don't want GUI asserts in retail builds either. Should we rather make the NoGuiOnAssert option retail?